File: 2562. Find the Array Concatenation Value.ts
function findTheArrayConcVal(nums: number[]): number {
    let concVal = 0;

    let start = 0;
    let end = nums.length - 1;
    while(start <= end){
        concVal += start === end ? nums[start] : concat(nums[start], nums[end]);
        ++start;
        --end;
    }

    return Number(concVal);
};

function concat(a: number, b: number) : number{
    return Number(`${a}${b}`);
};

File: 2562. Find the Array Concatenation Value.java
class Solution {
    public long findTheArrayConcVal(int[] nums) {
        long concVal = 0;

        int start = 0;
        int end = nums.length - 1;
        while(start <= end){
            concVal += start == end ? nums[start] : concat(nums[start], nums[end]);
            ++start;
            --end;
        }

        return concVal;
    }

    private long concat(long a, long b){
        return Long.parseLong(""+a+b);
    }
}
